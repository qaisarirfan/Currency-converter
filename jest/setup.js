jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');

  // The mock for `call` immediately calls the callback which is incorrect
  // So we override it with a no-op
  Reanimated.default.call = () => {};

  return Reanimated;
});

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useFocusEffect: () => jest.fn(),
    useRoute: () => jest.fn(),
    useNavigation: () => ({
      navigation: () => jest.fn(),
      canGoBack: () => jest.fn(),
      goBack: () => jest.fn(),
      push: () => jest.fn(),
    }),
  };
});

jest.mock('react-native-bootsplash', () => ({
  hide: jest.fn(),
  show: jest.fn(),
}));

jest.mock('react-native-localize', () => ({
  getLocales: () => [
    { countryCode: 'GB', languageTag: 'en-GB', languageCode: 'en', isRTL: false },
    { countryCode: 'US', languageTag: 'en-US', languageCode: 'en', isRTL: false },
    { countryCode: 'FR', languageTag: 'fr-FR', languageCode: 'fr', isRTL: false },
  ],

  getNumberFormatSettings: () => ({
    decimalSeparator: '.',
    groupingSeparator: ',',
  }),

  getCalendar: () => 'gregorian', // or "japanese", "buddhist"
  getCountry: () => 'US', // the country code you want
  getCurrencies: () => ['USD', 'EUR'], // can be empty array
  getTemperatureUnit: () => 'celsius', // or "fahrenheit"
  getTimeZone: () => 'Europe/Paris', // the timezone you want
  uses24HourClock: () => true,
  usesMetricSystem: () => true,

  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
}));

jest.mock('react-native-keyboard-aware-scroll-view', () => {
  const KeyboardAwareScrollView = require('react-native').ScrollView;
  return { KeyboardAwareScrollView };
});

jest.mock('react-native-gesture-handler', () => {
  // eslint-disable-next-line global-require
  const View = require('react-native/Libraries/Components/View/View');
  return {
    Swipeable: View,
    DrawerLayout: View,
    State: {},
    ScrollView: View,
    Slider: View,
    Switch: View,
    TextInput: View,
    ToolbarAndroid: View,
    ViewPagerAndroid: View,
    DrawerLayoutAndroid: View,
    WebView: View,
    NativeViewGestureHandler: View,
    TapGestureHandler: View,
    FlingGestureHandler: View,
    ForceTouchGestureHandler: View,
    LongPressGestureHandler: View,
    PanGestureHandler: View,
    PinchGestureHandler: View,
    RotationGestureHandler: View,
    /* Buttons */
    RawButton: View,
    BaseButton: View,
    RectButton: View,
    BorderlessButton: View,
    /* Other */
    FlatList: View,
    gestureHandlerRootHOC: jest.fn(),
    Directions: {},
  };
});

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
