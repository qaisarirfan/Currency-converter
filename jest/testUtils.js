import React from "react"
import {render} from "react-native-testing-library"
import {Provider} from "react-redux"
import configureStore from "redux-mock-store"
import ThemeContextProvider from "../src/ContextUtils/ThemeContext"
import ConversionContextProvider from "../src/ContextUtils/ConversionContext"
import {initialState as themesState} from "../src/redux/reducers/themes"

const INITIAL_STATE = {
  themes: themesState,
}
const mockStore = configureStore([])
const store = mockStore(INITIAL_STATE)

const reduxRender = (ui, options) =>
  render(ui, {
    wrapper: ({children}) => (
      <Provider store={store}>
        <ThemeContextProvider>
          <ConversionContextProvider>{children}</ConversionContextProvider>
        </ThemeContextProvider>
      </Provider>
    ),
    ...options,
  })

// re-export everything for convenience
export * from "react-native-testing-library"

// override render method
export {reduxRender as render, store}
