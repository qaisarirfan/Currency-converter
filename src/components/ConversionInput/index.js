import React, {useContext} from "react"
import {View, Text, TouchableOpacity, TextInput} from "react-native"
import PropTypes from "prop-types"
import themeStyles from "./styles"
import {ThemeContext} from "../../ContextUtils/ThemeContext"

// ConversionInput Component content
export const ConversionInput = ({text, onButtonPress, ...rest}) => {
  const {styleableTheme} = useContext(ThemeContext)

  const styles = themeStyles(styleableTheme)

  const containerStyles = [styles.container]

  if (rest.editable === false) {
    containerStyles.push(styles.containerDisabled)
  }
  return (
    <View style={containerStyles}>
      <TouchableOpacity style={styles.button} onPress={onButtonPress}>
        <Text style={styles.buttonText}>{text}</Text>
      </TouchableOpacity>
      <TextInput style={styles.input} {...rest} />
    </View>
  )
}

// ConversionInput Proptypes
ConversionInput.propTypes = {}

// ConversionInput Default props
ConversionInput.defaultProps = {}

export default ConversionInput
