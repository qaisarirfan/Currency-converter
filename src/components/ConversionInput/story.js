import React from "react"
import {storiesOf} from "@storybook/react-native"
import ConversionInput from "."
import {StoryWrapper} from "../../../storybook/StoryWrapper"

storiesOf("ConversionInput", module)
  .add("default", () => (
    <StoryWrapper>
      <ConversionInput text="test" onButtonPress={() => null} />
    </StoryWrapper>
  ))
  .add("readonly", () => (
    <StoryWrapper>
      <ConversionInput
        text="test"
        onButtonPress={() => null}
        editable={false}
      />
    </StoryWrapper>
  ))
