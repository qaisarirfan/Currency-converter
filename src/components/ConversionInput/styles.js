import {StyleSheet} from "react-native"
import Color from "color"
// ConversionInput: Styling here
const styles = (theme) => {
  return StyleSheet.create({
    container: {
      backgroundColor: "#FFF",
      marginVertical: 10,
      marginHorizontal: 20,
      flexDirection: "row",
      borderRadius: 5,
      borderWidth: StyleSheet.hairlineWidth,
      borderColor: theme[200],
    },
    containerDisabled: {
      backgroundColor: "#f0f0f0",
    },
    button: {
      padding: 15,
      borderRightColor: theme[100],
      borderRightWidth: StyleSheet.hairlineWidth,
      backgroundColor: "#FFF",
      borderTopLeftRadius: 5,
      borderBottomLeftRadius: 5,
    },
    buttonText: {
      fontSize: 18,
      color: "#000",
      fontWeight: "bold",
    },
    input: {
      flex: 1,
      padding: 10,
      fontSize: 16,
      color: Color(theme.A700).darken(0.6).hex(),
    },
  })
}
export default styles
