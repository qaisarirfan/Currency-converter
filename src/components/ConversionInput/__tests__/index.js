import React from "react"
import {render, cleanup, store} from '../../../../jest/testUtils';
import ConversionInput from ".."

afterEach(() => {
  cleanup();
  store.clearActions();
});

describe("<ConversionInput />", () => {
  const createProps = () => ({})
  it("Does not explode", () => {
    const props = createProps()
    const component = render(<ConversionInput {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
