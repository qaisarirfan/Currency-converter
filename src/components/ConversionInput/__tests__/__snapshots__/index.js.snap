// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConversionInput /> Does not explode 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFF",
        "borderColor": "#a7b6bd",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "marginHorizontal": 20,
        "marginVertical": 10,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderBottomLeftRadius": 5,
        "borderRightColor": "#cad3d7",
        "borderRightWidth": 0.5,
        "borderTopLeftRadius": 5,
        "opacity": 1,
        "padding": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    />
  </View>
  <TextInput
    allowFontScaling={true}
    rejectResponderTermination={true}
    style={
      Object {
        "color": "#00476E",
        "flex": 1,
        "fontSize": 16,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
